### ISSUE TITLE
Docker daemon crashes after a while while using wireguard container

### ISSUE THREAD

**Issue by petr-ujezdsky** [docker/for-mac#6585]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 6FB77661-8FDA-427A-926F-CF4FC04C2C6A/20221118105659

### Expected behavior
Docker daemon will not crash and the wireguard container keeps running

### Actual behavior
Docker daemon crashes when viewing YouTube video on connected device.

### Information

  - macOS Version: 12.6
  - Intel chip or Apple chip: Intel
  - Docker Desktop Version: 4.14.1
  - The problem appeared after upgrading from 4.12.0 to 4.14.0 and 4.14.1
  - I tried to turn off Virtualization framework which looked it helped. But now it crashed again.

### Output of `/Applications/Docker.app/Contents/MacOS/com.docker.diagnose check`
<details>
  <summary>diagnose check</summary>

```
 $ /Applications/Docker.app/Contents/MacOS/com.docker.diagnose check
Starting diagnostics

[PASS] DD0027: is there available disk space on the host?
[PASS] DD0028: is there available VM disk space?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[PASS] DD0017: can a VM be started?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0031: does the Docker API work?
[PASS] DD0013: is the $PATH ok?
[PASS] DD0003: is the Docker CLI working?
[PASS] DD0014: are the backend processes running?
[PASS] DD0007: is the backend responding?
[PASS] DD0008: is the native API responding?
[PASS] DD0009: is the vpnkit API responding?
[PASS] DD0010: is the Docker API proxy responding?
[PASS] DD0012: is the VM networking working?
[SKIP] DD0030: is the image access management authorized?
[PASS] DD0019: is the com.docker.vmnetd process responding?
[PASS] DD0033: does the host have Internet access?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[PASS] DD0017: can a VM be started?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0031: does the Docker API work?
[PASS] DD0032: do Docker networks overlap with host IPs?
segment 2022/11/18 12:24:09 ERROR: sending request - Post "https://api.segment.io/v1/batch": dial tcp [::]:443: connect: connection refused
segment 2022/11/18 12:24:09 ERROR: 1 messages dropped because they failed to be sent and the client was closed
No fatal errors detected.
```
</details>

### Steps to reproduce the behavior

  1. Start the modified `wg-easy` container (contains its own DNS server, see https://github.com/WeeJeWel/wg-easy/pull/323). Wireguard is VPN server.
  2. Use another device (eg. notebook) and connect using wireguard VPN to this container
  3. Run YouTube video on the connected device
  4. After a while (from few seconds to 2 minutes) the docker daemon crashes and the connected device loose internet connection (the container, server part of wireguard VPN, is not running anymore)

I see crashlogs in Console.app for `com.docker.vpnkit` but the text above suggests to *not* send it here. Please advise how to send them to you if you do not see them under the Diagnostics ID.
> Do *not* paste logfiles into github issues: upload diagnostics instead, which are
only visible to Docker engineers.

<details>
  <summary>The beginning of the log</summary>

```
-------------------------------------
Translated Report (Full Report Below)
-------------------------------------

Process:               com.docker.vpnkit [2207]
Path:                  /Applications/Docker.app/Contents/Resources/bin/com.docker.vpnkit
Identifier:            com.docker.vpnkit
Version:               ???
Code Type:             X86-64 (Native)
Parent Process:        com.docker.backend [665]
Responsible:           Docker [655]
User ID:               501

Date/Time:             2022-11-18 10:40:35.5643 +0100
OS Version:            macOS 12.6 (21G115)
Report Version:        12
Bridge OS Version:     6.6 (19P6067)
Anonymous UUID:        --redacted--


Time Awake Since Boot: 130000 seconds

System Integrity Protection: enabled

Crashed Thread:        3

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x00007fce28000000
Exception Codes:       0x0000000000000001, 0x00007fce28000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    Namespace SIGNAL, Code 11 Segmentation fault: 11
Terminating Process:   exc handler [2207]

VM Region Info: 0x7fce28000000 is not in any region.  Bytes after previous region: 1  Bytes before following region: 15728640
      REGION TYPE                    START - END         [ VSIZE] PRT/MAX SHRMOD  REGION DETAIL
      MALLOC_MEDIUM            7fce27800000-7fce28000000 [ 8192K] rw-/rwx SM=PRV  
--->  GAP OF 0xf00000 BYTES
      MALLOC_TINY              7fce28f00000-7fce29000000 [ 1024K] rw-/rwx SM=PRV  

```
</details>

**Comment by ackerthehacker2** [comment]
I think it has nothing to do with the container but the image.   Running  4.14.1 on os13.1.    Starts up, runs for about 5 min then stops.

Diag:  5C8E7B68-C878-42A0-8A7F-7811352B03F6/20221119071959

**Comment by petr-ujezdsky** [comment]
Does the container simply stops or does it take down the whole docker daemon as in my case?

Btw I have not updated the docker image. I use the same container since docker version 4.12.0 (or maybe even older).

**Comment by ackerthehacker2** [comment]
The whole daemon ....docker is not running .. I reverted back to 4.12.1 and it stays up for about 1-2 hrs before crashing.

**Comment by petr-ujezdsky** [comment]
I have reverted back to 4.12.0 (https://docs.docker.com/desktop/release-notes/#4120) and the container is now running without problem for 5 hours. I have watched some and was scrubbing through many youtube videos and other stuff.

It really looks like there is some problem in 4.14.x release.

**Comment by ackerthehacker2** [comment]
Docker desktop stops running …I went back to 4.12.0On Nov 19, 2022, at 6:08 AM, Petr Újezdský ***@***.***> wrote:Does the container simply stops or does it take down the whole docker daemon as in my case?Btw I have not updated the docker image. I use the same container since docker version 4.12.0 (or maybe even older).—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>

**Comment by LoganInTX** [comment]
Just tried 4.15 and it exhibits the same behavior. As soon as I try to download anything with SABNZBD then it crashes. Runs fine on 4.12 but none of the versions since.

**Comment by koraykoska** [comment]
Same for me in 4.15.0.

**Comment by koraykoska** [comment]
Just found this: https://github.com/docker/for-mac/issues/6530

I believe this was fixed in 4.16.0, you have to upgrade manually by checking updates in the Docker Desktop Settings.

**Comment by emaildanwilson** [comment]
possibly related, we're unable to establish multiple outbound wireguard tunnels from a container on 4.16.3 (M1 mac is all we tested) unless we disable `Use Virtualization framework` in the docker settings and restart. once we do that everything works great again.

**Comment by caroleefiallosswp60** [comment]
Diag: PRvrLeqbnwjgju9CPeTDlaBOTavZTfmWLeg0Lv9alxr.kFPZmFK6JX93lEgtjUr2lEPvmxnvis5YkhPvJbL2lUarlwPYJZ..!

**Comment by petr-ujezdsky** [comment]
Looks like this was resolved by #6530, thanks @koraykoska
